### PR TITLE
#2846: Fix bugs with the copy email button UI plus code cleanup - [RJM]

### LINKED ISSUE #2846: Django admin copy email button: display bugs
### Current Behavior
Currently, we have a copy button next to email fields on a few django admin pages. The domain request page has quite a few of these. However, some functionality has broke on them.

- Bug 1: The copy button does copy but after clicking it - the button displays the check mark and the text doesn't change back to "copied to clipboard". Basically, the button appears to "freeze".

- Bug 2: The copy button is displayed when no email exists for the given user. Since there is nothing to copy, this is counter-intuitive.

- Bug 3: The link coloration of the copy button in django dark mode has contrast issues. This is because it is not using the link-fg color.

### Expected Behavior
The copy button should behave as expected.

- Bug 1: The copy button should behave like the "copy email" buttons that currently exist under the other contacts table on the domain request view. 

- Bug 2: The copy button should not appear when no email exists.

- Bug 3: The copy button should be using the link-fg color.

### Steps to Reproduce

Bug 1:
1. Go to a domain request
2. Click the copy button under the creator field (next to their email)

Bug 2:
1. Go to a domain request
2. Modify the creator of the domain request, removing the "email" field from them
3. Go back to said domain request
4. Note the erroneous copy button

Bug 3:
1. Go to a domain request
2. Switch to dark mode (either system-wide or using the button next to log out in django admin)
3. Note the coloration of the button

### Environment

_No response_

### Additional Context

Regarding bug 1, this is breaking because the script that handles the copy logic expects there to be a `<span>` (i.e. `<span>Copy</span>`) inside the copy button.

This doesn't necessarily need to exist here, but that would be one solution to resolve.

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by rachidatecs**
## Ticket

Resolves #2846 

## Changes

- Fix color contrast bug
- Fix dynamic button label bug
- Hide button when no email
- Refactor admin-side button css (some cleanup)
- Fix bug with copy request summary (cc @abroddrick in case this resolves an existing ticket)
  - Remove submitter

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Comment by abroddrick** [comment]
@rachidatecs I can't see any bug regarding the copy request button, so you are good!

File: src/registrar/assets/sass/_theme/_admin.scss [hunk 1]
```diff
@@ -425,6 +428,37 @@ button .usa-icon,
...
+
+    span {
+        padding-left: 0.05rem;
+    }
+
+}
+.usa-button__small-text,
+.usa-button__small-text span {
+    font-size: 13px;
+}
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/assets/sass/_theme/_admin.scss:461]: Matching DJA and I don't feel like bending over backwards for it.

File: src/registrar/templates/admin/input_with_clipboard.html [hunk 2]
```diff
@@ -17,23 +17,27 @@
...
-<div class="admin-icon-group admin-icon-group__clipboard-link">
+<div class="admin-icon-group">
     <input aria-hidden="true" class="display-none" value="{{ field.email }}" />
-    <button 
-    class="usa-button usa-button--unstyled padding-right-1 usa-button--icon button--clipboard copy-to-clipboard text-no-underline padding-left-05" 
-    type="button"
-    >
-        <svg 
-        class="usa-icon" 
+    {% if field.email is not None %}
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/templates/admin/input_with_clipboard.html:28]: I dunno. Does that invisible input serve any functions outside of this file? I've been burned before lol

File: src/registrar/templates/django/admin/includes/detail_table_fieldset.html [hunk 3]
```diff
@@ -254,7 +254,7 @@ <h2 class="usa-modal__heading">
                     <table>
                         <thead>
                             <tr>
-                                <th colspan="4">Other contact information</th>
+                                <th colspan="5">Other contact information</th>
```
[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/templates/django/admin/includes/detail_table_fieldset.html:257]: That would be an accessibility issue, since the body rows do not have ths (that td with the button would not have any parent th anywhere)

[the remark this answers is not in the document]
Comment (rachidatecs) [reply src/registrar/templates/django/admin/includes/detail_table_fieldset.html:257]: Gonna go ahead and merge

File: src/registrar/admin.py [hunk 4]
```diff
@@ -3104,6 +3104,83 @@ def get_non_admin_users(self, obj):
 
         return non_admin_users
 
+    def display_admins(self, obj):
```
Comment (rachidatecs) [inline src/registrar/admin.py:3107]: Who removed/why was this removed from main? This code came back when I merged main. cc @abroddrick @lizpearl

Comment (rachidatecs) [reply src/registrar/admin.py:3107]: Actually, I think it's a @zandercymatics refactor

Comment (rachidatecs) [reply src/registrar/admin.py]: Got it
